app/models/api_authenticator.rb

```diff
@@ -0,0 +1,42 @@
+class ApiAuthenticator < Struct.new(:controller)
```

Comment:
When we reorganize things, these should probably become Policy models (I had scoped them before to only be accessible from controllers because I wanted it to be clear)